src/doom/r_plane.c

```diff
@@ -484,7 +484,17 @@ void R_DrawPlanes (void)
 
 		if ((unsigned) dc_yl <= dc_yh) // [crispy] 32-bit integer math
 		{
-		    angle = ((an + xtoviewangle[x])^flip)>>ANGLETOSKYSHIFT;
+			// [crispy] Draw skies horizontally linear.
+            // Taken from GZDoom and refactored for integer math.
+			if (crispy->linearsky) 
+			{
+                angle_t xangle = (angle_t)((viewwidth/2 - x) * (ANG90/NONWIDEWIDTH));
```

Comment:
I refactored the equation and it works fine, with no floating point math. I am assuming that `viewwidth` is always the same as `SCREENWIDTH`, though. Is this the case? Is it better to use one over the other? (also yes tab formatting again, sorry, will correct later.)

Reply:
Ah, never mind, `viewwidth` decreases when the screensize isn't max.